src/sobj.c

```diff
@@ -0,0 +1,113 @@
+/* SOBJ -- autonomous shared-object reference handling.
+ *
+ * Copyright (c) 2017-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of (a) the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+ */
+
+#include "server.h"
+#include "sobj.h"
+
+void sobj_type_destructor(dict *d, void *key)
+{
+    UNUSED(d);
+    decrRefCount(key);
+    server.stat_shared_objects_cnt--;
+}
+
+dictType sobj_dict_type = {
+    dictObjHash,
+    NULL,
+    NULL,
+    dictObjKeyCompare,
+    sobj_type_destructor,
+    NULL,
+    .no_value = 1, /* Implement a set (no values). Allows optimizing out the dictEntry struct */
+    .keys_are_odd = 0
+};
+
+dict *sobj_init(void)
```

Comment:
please create a clear seperation between pool-related functions and obj-related functions